src/codegen/compile.ml

```diff
@@ -5557,16 +5705,16 @@ module MakeSerialization (Strm : Stream) = struct
         write_text env get_data_buf get_x
       | Func _ ->
         write_byte env get_data_buf (compile_unboxed_const 1l) ^^
-        get_x ^^ Arr.load_field 0l ^^ write env (Obj (Actor, [])) ^^
-        get_x ^^ Arr.load_field 1l ^^ write env (Prim Text)
+        get_x ^^ Arr.load_field env 0l ^^ write env (Obj (Actor, [])) ^^
+        get_x ^^ Arr.load_field env 1l ^^ write env (Prim Text)
```

Comment:
Duplicated forwarding (if we care).

Reply:
We could use this logic. But I am not sure whether there is a performance gain (or even degrade as for `Arr.idx`). 
```
        get_x ^^ Tagged.load_forwarding_pointer env ^^ set_x ^^
        inc_data_size (compile_unboxed_const 1l) ^^ (* one byte tag *)
        get_x ^^ Heap.load_field Int32.(add 0l Arr.header_size) ^^ size env (Obj (Actor, [])) ^^
        get_x ^^ Heap.load_field Int32.(add 1l Arr.header_size) ^^ size env (Prim Text)
```

Reply:
Probably not worth it then. This code is rarely used (higher-order Candid)